packages/apps/reputation-oracle/server/src/modules/nda/nda.controller.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Req,
+  UseFilters,
+  UseGuards,
+} from '@nestjs/common';
+import { NDAService } from './nda.service';
+import {
+  ApiBearerAuth,
+  ApiBody,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { JwtAuthGuard } from 'src/common/guards';
+import { RequestWithUser } from 'src/common/interfaces/request';
+import { NDAErrorFilter } from './nda.error.filter';
+import { AuthConfigService } from 'src/config/auth-config.service';
+import { NDASignatureDto } from './nda.dto';
+
+@ApiTags('NDA')
+@UseFilters(NDAErrorFilter)
+@ApiBearerAuth()
+@UseGuards(JwtAuthGuard)
+@Controller('nda')
+export class NDAController {
+  constructor(
+    private readonly ndaService: NDAService,
+    private readonly authConfigService: AuthConfigService,
+  ) {}
+
+  @ApiOperation({
+    summary: 'Retrieves latest NDA URL',
+    description:
+      'Retrieves the latest NDA URL that users must sign to join the oracle',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'URL retrieved successfully',
+    type: String,
+  })
+  @ApiResponse({
```

Comment:
Let's please remove this api response: we decided to not document `401` since it's common for api